sig-network/api-extensions-best-practices.md

```diff
@@ -0,0 +1,145 @@
+# API Extensions - Best Practices
+
+It is sometimes desirable to build new networking features as [CRDs] rather
+than [in-tree]. Reasons may include (but aren't limited to):
+
+* A desire to be free of lockstep with the [KEP] process
+* Experimentation and highly iterative development
+* The feature is useful to multiple parties, but is niche
+
+Whatever the case may be, we have multiple SIG Network sub-projects that have
+taken this approach for various reasons:
+
+* [Gateway API]
+* [Network Policy]
+* [Multi Network]
+
+This document intends to share guidelines and best practices for how to create
+and manage these kinds of "official, but add-on and optional" features, which
+throughout this document we'll refer to as **Official CRDs**.
+
+> **Note**: **This is not a standard**, but **projects should try and follow
+> this guidance** unless they have some good reasons not to, as the guidance
+> here comes from the experience of those who have walked this road before.
+
+> **Note**: This is **not a comprehensive guide**.
+
+[CRDs]:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/
+[in-tree]:https://github.com/kubernetes/kubernetes/
+[KEP]:https://github.com/kubernetes/enhancements/blob/master/keps/README.md
+[Gateway API]:https://github.com/kubernetes-sigs/gateway-api
+[Network Policy]:https://github.com/kubernetes-sigs/network-policy-api
+[Multi Network]:https://github.com/kubernetes-sigs/multi-network
+
+## Objectives
+
+At a high level, SIG Network sub-projects that deploy as [CRDs] will need to
+deal with (at least) the following dimensions:
+
+* Enhancement Proposals
+* Developing CRDs
+* Delivering CRDs
+* Conformance Levels
+* Conformance Tests & Reports
+* CRD Lifecycle Management
+* Documentation
+* Community & Evangelism
```

Comment:
"Community & Evangelism" seems more of a WG thing in general than a CRD vs non-CRD thing.

Reply:
Removed.